XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -766,20 +804,55 @@ public VirtualisationStatus GetVirtualisationStatus
 
                 if (vm_guest_metrics == null || !vm_guest_metrics.PV_drivers_installed)
                 {
-                    return VirtualisationStatus.PV_DRIVERS_NOT_INSTALLED;
+                    return VirtualisationStatus.PV_DRIVERS_NOT_INSTALLED | GetNewVirtualisationFlags();
                 }
                 else if (!vm_guest_metrics.PV_drivers_up_to_date)
                 {
-                    return VirtualisationStatus.PV_DRIVERS_OUT_OF_DATE;
+                    return VirtualisationStatus.PV_DRIVERS_OUT_OF_DATE | GetNewVirtualisationFlags();
                 }
                 else
                 {
-                    return VirtualisationStatus.OPTIMIZED;
+                    if (!IsNewVM)
+                        return VirtualisationStatus.OPTIMIZED;

```

Comment:
Maybe you need a bit more dev testing, which could have caught that some old-style VMs didn't get some operations they should have got.